### PR TITLE
SingleRange Begin/End needs to be exposed for GrainServices

### PR DESCRIPTION
**Pull request by jamescarter-le**
Added ISingleRange interface to be exposed externally. Also exposed helper method NextTimeSpan from StandardExtensions as well as some helper methods in RangeFactory.

### CODE REVIEW

**Comment by jamescarter-le** [comment]
Can be included in a 1.4.X if one of those are going out.

File: src/Orleans/Utils/StandardExtensions.cs [hunk 1]
```diff
@@ -39,15 +39,21 @@ public static TimeSpan Min(TimeSpan first, TimeSpan second)
             return first < second ? first : second;
         }
 
-        public static TimeSpan NextTimeSpan(this SafeRandom random, TimeSpan timeSpan)
+        public static TimeSpan NextTimeSpan(TimeSpan timeSpan)
+        {
+            var random = new SafeRandom();
```
Comment (sergeybykov) [inline src/Orleans/Utils/StandardExtensions.cs]: This line would allocate on every call. Can we avoid that?

**Comment by jamescarter-le** [comment]
We could move it static to the class. Otherwise we could make SafeRandom
public so people could keep their own instance. I wanted to expose the
NextTimespan method without exposing anything further.

**Comment by sergeybykov** [comment]
In general, we've been reluctant to expose public extension methods for BCL types, so that we don't appear to be extending functionality of those types: `TimeSpan`, `Task`, etc. The concern is that if similar functionality is added at some point to BCL, our extensions will look odd. Of course, we violated that general rule where we deemed it necessary.

In that light, I'm wondering if it's really necessary to expose `StandardExtensions` and `SafeRandom` as part of the Orleans API surface.

**Comment by jamescarter-le** [comment]
Happy to look to modify the way the random time spans can be generated for
grain services and either provide a helper method inside GrainService or
just leave the user to their own devices. This is for refreshing of the
reminder table/user specific implementations.

**Comment by sergeybykov** [comment]
Let's try that approach then. Thanks!

**Comment by jamescarter-le** [comment]
@sergeybykov I have removed the change to expose the Timer extension method.

**Comment by sergeybykov** [comment]
Thank you, @jamescarter-le! Sorry about delay.

**Pull request merged**